scheduler/src/cook/mesos/api.clj

```diff
@@ -403,21 +452,16 @@
 ;;; On GET; use repeated job argument
 (defn job-resource
   [conn fid task-constraints gpu-enabled?]
-  (-> (resource
-        :available-media-types ["application/json"]
-        :allowed-methods [:post :get :delete]
-        :malformed? (fn [ctx]
-                      (condp contains? (get-in ctx [:request :request-method])
-                        #{:get :delete}
-                        (if-let [jobs (get-in ctx [:request :params "job"])]
-                          (let [jobs (if-not (vector? jobs) [jobs] jobs)]
-                            (try
-                              [false {::jobs (mapv #(UUID/fromString %) jobs)}]
-                              (catch Exception e
-                                [true {::error e}])))
-                          [true {::error "must supply at least one job query param"}])
+  (-> (liberator/resource
+       :available-media-types ["application/json"]
+       :allowed-methods [:post :get :delete]
+       :malformed? (fn [ctx]
+                     (condp contains? (get-in ctx [:request :request-method])
+                       ;; schema will already validate.
```

Comment:
(Basically, if the coercion fails, you'll get a 400 response with a JSON structure describing why the coercion failed).